lib/ScssCompiler.js

```diff
@@ -0,0 +1,280 @@
+const _ = require('lodash');
+const path = require('path');
+const nodeSassFork = require('@bigcommerce/node-sass');
+const dartSass = require('sass');
+const Fiber = require('fibers');
+const { promisify } = require('util');
+
+class ScssCompiler {
+    /**
+     * WARNING! Make sure to create a new instance per each compilation to avoid processes interfering each other
+     *
+     * @param {object} logger
+     * @param {object} engines
+     * @param {object} engines.primary
+     * @param {object} engines.fallback
+     * @param {function} logger.error
+     */
+    constructor(
+        logger,
+        engines = {
```

Comment:
Do we really need this ? I think we should just do something like `activeEngine = dartSass` here and then in the catch we can switch it over to `nodeSassFork`

Reply:
I specified it in the constructor so that we can test it easily passing a proper mock

Reply:
so if you set in the constructor then in `nodeSassFork` we override it with `dartSass`. So even if you set `nodeSassFork` it will be overridden in the try before it fails and goes into catch